### PR TITLE
Fix: modal dependency

### PR DESCRIPTION
**Pull request by ganchoradkov**
## Description
PR to make `@walletconnect/modal` an explicit dependency of ethereum provider

## Type of change

- [x] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Draft PR (breaking/non-breaking change which needs more work for having a proper functionality _[Mark this PR as ready to review only when completely ready]_)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How has this been tested?
dogfooding - `2.10.4-canary-f2d9992b`

## Checklist

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

# Additional Information (Optional)

Please include any additional information that may be useful for the reviewer.

### CODE REVIEW

**Review by mohamedabdurahmanhassa — COMMENTED** [review]
88baade717f008c4d8bf9213f855eadda2cd0bba2aa560363719c0ddcee540a3

File: providers/ethereum-provider/package.json [hunk 1]
```diff
@@ -50,22 +50,14 @@
     "@walletconnect/sign-client": "2.10.4",
     "@walletconnect/types": "2.10.4",
     "@walletconnect/universal-provider": "2.10.4",
+    "@walletconnect/modal": "^2.4.3",
```
Comment (ignaciosantise) [inline providers/ethereum-provider/package.json:53]: why this change?

Comment (ganchoradkov) [reply providers/ethereum-provider/package.json:53]: the modal is moved to an actual dependency so its always downloaded togheder with Ethereum provider thus resolving the build issues

Comment (ignaciosantise) [reply providers/ethereum-provider/package.json:53]: will this bring versioning issues with other packages that use another version of the modal? Wagmi for example has a fixed version the modal -> [here](https://github.com/wagmi-dev/wagmi/blob/eb06c91b6b1b27c6b8283060bdbd6f013bbf86ef/packages/connectors/package.json#L27)

build issues doesn't solve if people install the modal dependency following our docs?
`yarn add @walletconnect/ethereum-provider @walletconnect/modal`

Comment (ganchoradkov) [reply providers/ethereum-provider/package.json:53]: Any patch version would be fine

**Review by ignaciosantise — APPROVED** [review]

**Review by glitch-txs — APPROVED** [review]
